tests/Google/Ads/GoogleAds/Util/V6/ResourceNamesTest.php

```diff
@@ -2403,6 +2427,31 @@ public function testGetNameForTopicConstant()
         $this->assertEquals($topicId, $names['topic_id']);
     }
 
+    /**
+     * @covers \Google\Ads\GoogleAds\Util\V6\ResourceNames::forTopicView()
+     */
+    public function testGetNameForTopicView()
+    {
+        $adGroupId = 111111;
+        $criterionId = 222222;
+        $expectedResourceName = sprintf(
+            'customers/%s/topicViews/%s~%s',
+            self::CUSTOMER_ID,
+            $adGroupId,
+            $criterionId
+        );
+        $this->assertEquals($expectedResourceName, ResourceNames::forTopicView(
+            self::CUSTOMER_ID,
+            $adGroupId,
+            $criterionId
+        ));
+
+        $names = TopicViewServiceClient::parseName($expectedResourceName);
+        $this->assertEquals(self::CUSTOMER_ID, $names['customer_id']);
+        $this->assertEquals($adGroupId, $names['ad_group_id']);
+        $this->assertEquals($criterionId, $names['criterion_id']);
```

Comment:
So, the TopicViewServiceClient generates ad_group_id and criterion_id for us?

Reply:
I believe so, would not the assert fail if that is not the case? What makes you think that there is something wrong?

Reply:
The new scheme sometimes parse a name and generates tokens that are close to the resource name, e.g., 'customer_id' and 'topic_view'.
This case doesn't follow that scheme though.